frontend/packages/dev-console/src/components/edit-application/EditApplication.tsx

```diff
@@ -1,113 +1,105 @@
-    docker: {
-      dockerfilePath: _.get(
-        appResources,
-        'buildConfig.data.spec.strategy.dockerStrategy.dockerfilePath',
-        'Dockerfile',
-      ),
-      containerPort: parseInt(
-        _.split(_.get(appResources, 'route.data.spec.port.targetPort'), '-')[0],
-        10,
-      ),
-    },
-    image: {
-      selected: currentImage[0] || '',
-      recommended: '',
-      tag: currentImage[1] || '',
-      tagObj: {},
-      ports: [],
-      isRecommending: false,
-      couldNotRecommend: false,
-    },
-    route: EditApplicationUtils.getRouteData(_.get(appResources, 'route.data'), editAppResource),
-    resources: EditApplicationUtils.getResourcesType(editAppResource),
-    serverless: EditApplicationUtils.getServerlessData(editAppResource),
-    pipeline: {
-      enabled: false,
-    },
-    build: EditApplicationUtils.getBuildData(_.get(appResources, 'buildConfig.data')),
-    deployment: EditApplicationUtils.getDeploymentData(editAppResource),
-    labels: EditApplicationUtils.getUserLabels(editAppResource),
-    limits: EditApplicationUtils.getLimitsData(editAppResource),
+  const handleRedirect = (project: string) => {
+    const perspectiveData = plugins.registry
+      .getPerspectives()
+      .find((item) => item.properties.id === perspective);
+    const redirectURL = perspectiveData.properties.getImportRedirectURL(project);
+    history.push(redirectURL);
   };
-
   const handleSubmit = (values, actions) => {
-    const imageStream =
-      values.image.selected && builderImages ? builderImages[values.image.selected].obj : null;
-
-    createOrUpdateResources(
-      values,
-      imageStream,
-      false,
-      false,
-      'update',
-      appResources,
-      editAppResource,
-    )
-      .then(() => {
-        actions.setSubmitting(false);
-        actions.setStatus({ error: '' });
-        onSubmit();
-      })
-      .catch((err) => {
-        actions.setSubmitting(false);
-        actions.setStatus({ error: err.message });
-      });
+    if (values.build.strategy) {
+      const imageStream =
+        values.image.selected && builderImages ? builderImages[values.image.selected].obj : null;
+      createOrUpdateGitResources(values, imageStream, false, false, 'update', appResources)
+        .then(() => {
+          actions.setSubmitting(false);
+          actions.setStatus({ error: '' });
+          handleRedirect(namespace);
+        })
+        .catch((err) => {
+          actions.setSubmitting(false);
+          actions.setStatus({ error: err.message });
+        });
+    } else {
+      createOrUpdateDeployImageResources(values, false, 'update', appResources)
+        .then(() => {
+          actions.setSubmitting(false);
+          actions.setStatus({ error: '' });
+          handleRedirect(namespace);
+        })
+        .catch((err) => {
+          actions.setSubmitting(false);
+          actions.setStatus({ error: err.message });
+        });
+    }
```

Comment:
you could write something like this to avoid repeating the onSubmit Handler logic. 
```
const onFormSubmit = () =>
      Promise.resolve()
        .then(() => {
          actions.setSubmitting(false);
          actions.setStatus({ error: '' });
          handleRedirect(namespace);
        })
        .catch((err) => {
          actions.setSubmitting(false);
          actions.setStatus({ error: err.message });
        });
    if (!_.isEmpty(values.build.strategy)) {
      const imageStream =
        values.image.selected && builderImages ? builderImages[values.image.selected].obj : null;
      createOrUpdateResources(values, imageStream, false, false, 'update', appResources)
      .then(onFormSubmit);
    } else {
      createOrUpdateDeployImageResources(values, false, 'update', appResources)
        .then(onFormSubmit)
};
```